### PR TITLE
backporting fix to ethsign from symbolic

### PR DESCRIPTION
**Pull request by realisation**
Wholesale copy of default.nix and src/ethsign from the symbolic branch. 

Installed with following commands from a fresh clone after replacing default.nix and src/ethsign: 
```
git submodule init
git submodule update --recursive 
nix-env -i ethsign -f . 
```

### CODE REVIEW

File: src/ethsign/CHANGELOG [hunk 1]
```diff
@@ -4,15 +4,6 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
-## [0.14.0] - 2020-05-29
```
Comment (asymmetric) [inline src/ethsign/CHANGELOG:7]: Why are we removing this?

If the changes have been reverted, we should mention that instead.

**Comment by MrChico** [comment]
Probably better to cherry-pick the relevant commits, because there was a recent change merged to master (#378 )

**Comment by MrChico** [comment]
I also found this nixpkgs bump to be less dramatic when we keep the old nixpkgs for `geth`. See the `oldnixpkgs` related sections of `overlay.nix` on the symbolic branch

**Comment by asymmetric** [comment]
What's the relationship between the nixpkgs bump and the changes in ethsign? Could we do them saparately (and maybe bump to 20.03)?

**Comment by MrChico** [comment]
I think the geth libraries on which ethsign depends are updated along with the nixpkgs bump, and there were some minor changes needed. Could also go with 20.03. This is mainly to address #383

**Comment by realisation** [comment]
I just cherry picked the change to ethsign from the symbolic branch and updated default.nix manually since that commit was bundled up with many other changes on the symbolic branch.

I thought about bumping nixpkgs to 20.3, but that appears to involve a change to "rev" in default.nix and I'm not sure where to retrieve that. 

What is that all about?

**Comment by realisation** [comment]
I just cherry picked the change to ethsign from the symbolic branch and updated default.nix manually since that commit was bundled up with many other changes on the symbolic branch.

I thought about bumping nixpkgs to 20.3, but that appears to involve a change to "rev" in default.nix and I'm not sure where to retrieve that. 

What is that all about?

**Comment by MrChico** [comment]
I think we can do the bump to 20.03 later!

**Comment by MrChico** [comment]
Sweet! Do you think this needs a version bump @asymmetric ?

**Comment by MrChico** [comment]
Thanks for the fix @realisation !

**Pull request merged**